File: server/src/database/entities/notification.rs
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize, sqlx::Type)]
#[serde(rename_all(serialize = "camelCase", deserialize = "snake_case"))]
#[sqlx(type_name = "noti_type", rename_all = "snake_case")]
pub enum NotificationType {
    LikePost,
    LikeDiscussion,
    LikeCommentBlog,
    LikeCommentPost,
    LikeCommentDiscussion,
    CommentBlog,
    CommentPost,
    CommentDiscussion,
    CommentTagBlog,
    CommentTagPost,
    CommentTagDiscussion,
    GameDiscussion,
    Subscribe,
    UserAddedGame,
    UserUpdatedGame,
    UserPost,
    VoteGame,
}

File: web/src/models/notification.ts
export type NotificationType =
  | 'likePost'
  | 'likeDiscussion'
  | 'likeCommentBlog'
  | 'likeCommentPost'
  | 'likeCommentDiscussion'
  | 'commentBlog'
  | 'commentPost'
  | 'commentDiscussion'
  | 'commentTagBlog'
  | 'commentTagPost'
  | 'commentTagDiscussion'
  | 'gameDiscussion'
  | 'subscribe'
  | 'userAddedGame'
  | 'userUpdatedGame'
  | 'userPost'
  | 'voteGame';

export type Notification = {
  id: number;
  toUserId: number;
  byUserId: number;
  byUserName: string;
  byUserNameAvatar: string;
  byObjectId: string;
  targetType: NotificationType;
  targetId: string;
  parentTargetId: string;
  isRead: boolean;
  createdAt: string;
};
